### ISSUE TITLE
Inline ignore messages

### ISSUE THREAD

**Issue by s0undt3ch** [koalaman/shellcheck#145]
There should be a way to disable some checks by their codes in-line:

``` bash
if ( some sh scripting ); then    # shellcheck-ignore: <CODE TO IGNORE>
```

**Comment by gdbtek** [comment]
does anyone know if it's possible to disable error checking for a file?

**Comment by koalaman** [comment]
@gdbtek There's currently no way to make shellcheck ignore a file it's been explicitly asked to check. How are you running into this issue?

**Comment by gdbtek** [comment]
@koalaman , I have a case that I think it would be very handy to ignore a file. For example for the case of configuration files:

```
$ cat config.sh
var1='123'
var2='456'

$ cat app.sh
source config.sh

echo "${var1}"
echo "${var2}"

$ shellcheck config.sh app.sh 

In config.sh line 3:
var1='123'
^-- SC2034: var1 appears unused. Verify it or export it.

In config.sh line 4:
var2='456'
^-- SC2034: var2 appears unused. Verify it or export it.
```

this is just a quick sample. In my large app, I have many config files that will be shared by multiple apps. So when I ran shellcheck, I have a lot of warning of `SC2034` from all config files. That's why I think it would be nice to have a way to ignore files. Would be nice if:

1) shellcheck support some sort of `.shellcheckignore` file which is very similiar to `.jshintignore` or `.eslintignore` file in NodeJS code quality tool.
2) A directive that can be inserted on to of shell script to ignore the entire file.

For now, I have workaround by writing a short shell script to ignore them.

Thanks

**Comment by tgrey1** [comment]
@gdbtek - try this, it is how I disable a specific warning for an entire file.

I use a directive, and the braces create a scope for it, which includes the rest of the source.  This has worked fine for me in simple one off scripts, but I'm not sure of the implications with included files.  Worth a shot!

```
$ cat config.sh
#!/bin/bash -e
# shellcheck disable=SC2034
{
var1='123'
var2='456'
}
```

**Comment by btilford** [comment]
What if your linting a directory that is 99% shell scripts but 1 file isn't. Is there any way to ignore that 1 file assuming directives can't be used because they are invalid syntax within that file?

**Comment by undergroundwires** [comment]
Would you consider supporting disabling `inline` as in the same line? So instead of next line, using the same line:

Do this:

```sh
readonly SELF_DIRECTORY=$(dirname "$0") # shellcheck disable=SC2155
``

Instead of this:

```sh
# shellcheck disable=SC2155
readonly SELF_DIRECTORY=$(dirname "$0") 
``

It would allow writing cleaner code without shellcheck directives breaking the logical flow.

Let me know if I should create different issue.

**Comment by brother** [comment]
@undergroundwires see recent comment on issue https://github.com/koalaman/shellcheck/issues/1295#issuecomment-955789131
